jcl/src/java.base/share/classes/jdk/internal/ref/CleanerShutdown.java

```diff
@@ -49,25 +49,31 @@ public static void shutdownCleaner() {
 		
 		try {
 			Method phantomRemove = PhantomCleanable.class.getDeclaredMethod("remove", (Class<?>[]) null); //$NON-NLS-1$
+			/*[IF !Java16]*/
 			Method weakRemove = WeakCleanable.class.getDeclaredMethod("remove", (Class<?>[]) null); //$NON-NLS-1$
 			Method softRemove = SoftCleanable.class.getDeclaredMethod("remove", (Class<?>[]) null); //$NON-NLS-1$
+			/*[ENDIF]*/
 			AccessController.doPrivileged((PrivilegedAction<Void>) () -> {
 				phantomRemove.setAccessible(true);
+				/*[IF !Java16]*/
 				weakRemove.setAccessible(true);
 				softRemove.setAccessible(true);
+				/*[ENDIF]*/
 				return null;
 			});
 			while(!commonCleanerImpl.phantomCleanableList.isListEmpty()) {
 				phantomRemove.invoke(commonCleanerImpl.phantomCleanableList, (Object[]) null);
 			}
 
+			/*[IF !Java16]*/
 			while(!commonCleanerImpl.weakCleanableList.isListEmpty()) {
 				phantomRemove.invoke(commonCleanerImpl.weakCleanableList, (Object[]) null);
```

Comment:
I think this should use `weakRemove` rather than `phantomRemove`.

Reply:
I think we should just delete the weak and soft code for all versions considering it has never worked properly. Thoughts?

Reply:
That seems reasonable: the broken code (if it ever executed) would throw `IllegalArgumentException`.